compiler/rustc_errors/src/lib.rs

```diff
@@ -1719,19 +1719,17 @@ impl HandlerInner {
             match (
                 self.err_count() + self.lint_err_count,
                 self.delayed_bug_count(),
-                self.flags.treat_err_as_bug.map(|c| c.get()).unwrap_or(0),
+                self.flags.treat_err_as_bug.map(|c| c.get()).unwrap(),
```

Comment:
If `treat_err_as_bug` is unwrapped here, then it should probably not be an option.

Reply:
This is guarded behind an `if self.treat_err_as_bug()` [which currently implies that the flag is `Some`](https://github.com/rust-lang/rust/blob/66ab7e6883c563ceacf8386a50e851e9855cd6b4/compiler/rustc_errors/src/lib.rs#L1439-L1443), so the unwrap is infallible.

And I think `Option<NonZeroUsize>` is more natural than an `usize`, as it's clearly 1-indexed.